### ISSUE TITLE
lib{asm,caml}run_shared.so and lib{asm,caml}run_pic.a not installed with 4.08

### ISSUE THREAD

**Issue by hhugo** [ocaml/ocaml#8561]
```
$ find ~/.opam/4.07.1/lib/ocaml/ -name "lib*run*"
/home/hugo/.opam/4.07.1/lib/ocaml/libcamlrun_shared.so
/home/hugo/.opam/4.07.1/lib/ocaml/libasmrund.a
/home/hugo/.opam/4.07.1/lib/ocaml/libasmrun_shared.so
/home/hugo/.opam/4.07.1/lib/ocaml/libcamlruni.a
/home/hugo/.opam/4.07.1/lib/ocaml/libasmruni.a
/home/hugo/.opam/4.07.1/lib/ocaml/libcamlrun.a
/home/hugo/.opam/4.07.1/lib/ocaml/libcamlrund.a
/home/hugo/.opam/4.07.1/lib/ocaml/libcamlrun_pic.a
/home/hugo/.opam/4.07.1/lib/ocaml/libasmrun_pic.a
/home/hugo/.opam/4.07.1/lib/ocaml/libasmrunp.a
/home/hugo/.opam/4.07.1/lib/ocaml/libasmrun.a
```
vs
```
$ find _opam/lib/ocaml/ -name "lib*run*"
_opam/lib/ocaml/libasmrund.a
_opam/lib/ocaml/libcamlruni.a
_opam/lib/ocaml/libasmruni.a
_opam/lib/ocaml/libcamlrun.a
_opam/lib/ocaml/libcamlrund.a
_opam/lib/ocaml/libasmrunp.a
_opam/lib/ocaml/libasmrun.a
```

**Comment by hhugo** [comment]
cc @xclerc

**Comment by gasche** [comment]
Are both those installation produced from the default configuration options? If so, then not installing the files anymore is a regression (possibly due to configure-related changes) that should be fixed. But I'm not sure which configuration options were used for your 4.07.1 installation.

**Comment by hhugo** [comment]
4.07.1 comes from a standard opam install.

4.08 was installed with a plain `./configure`.
It seems that the install is controlled by `SUPPORTS_{BYTECODE,NATIVE}_SHARED_LIBRARIES`
but only `SUPPORTS_SHARED_LIBRARIES` is defined.

**Comment by hhugo** [comment]
https://github.com/janestreet/ocaml/commit/d628cb62580c01a5b16b54d5b532eb35b98a1e32 fixes the issue but might not be the right fix.

**Comment by dra27** [comment]
At a glance, the JS fix looks like it would fail when native code isn’t built.

**Comment by xclerc** [comment]
@dra27 Why would it? `NATIVE_STATIC_LIBRARIES` is always
[non-empty](https://github.com/janestreet/ocaml/blob/d628cb62580c01a5b16b54d5b532eb35b98a1e32/runtime/Makefile#L66), and `NATIVE_XYZ_LIBRARIES` seem to be used
only when the `opt` target is invoked.

**Comment by dra27** [comment]
@xclerc - too quick a glance from me, apparently! It is indeed that this controls the building of lists of libraries to build which are then used elsewhere, so the bytecode-only (or nativecode-only) paths would be fine.

As @gasche has already added, this is a small slip in #1852 - that JS commit does indeed fix the issue. We should allow @shindere a chance to respond, it being the weekend, but it looks as though those variables were introduced when `asmrun/Makefile` and `byterun/Makefile` were merged together to preserve separate meanings and then they just weren't renamed back later in the process when it realised they were identical.

**Comment by hhugo** [comment]
with janestreet@d628cb6, the genrated lib/ocaml/libasmrun_pic.a is almost empty

**Comment by xclerc** [comment]
@hhugo Indeed; the `libasmrunpic_OBJECTS` used [here](https://github.com/ocaml/ocaml/blob/4.08/runtime/Makefile#L326) contains
only `amd64_libasmrunpic.o`, while it should presumably contains
`*pic.o`.

**Comment by xclerc** [comment]
This is caused by the following typo:
```
--- a/runtime/Makefile
+++ b/runtime/Makefile
@@ -115,7 +115,7 @@ libasmruni_OBJECTS := $(NATIVE_C_SOURCES:.c=_ni.$(O)) $(ASM_OBJECTS)
 libasmrunp_OBJECTS := $(NATIVE_C_SOURCES:.c=_np.$(O)) \
   $(ASM_OBJECTS:.$(O)=_libasmrunp.$(O))
 
-libasmrunpic_OBJECTS := $(NATIVE_C_SOURCESS:.c=_npic.$(O)) \
+libasmrunpic_OBJECTS := $(NATIVE_C_SOURCES:.c=_npic.$(O)) \
   $(ASM_OBJECTS:.$(O)=_libasmrunpic.$(O))
 
 # General (non target-specific) assembler and compiler flags
```

**Comment by xclerc** [comment]
For the record, I just run a full build, with `--warn-undefined-variables`
passed to `make`. Here are the undefined variables (after filtering variables
that are "rightly" undefined - `otherlibs/Makefile.otherlibs.common`
expects some variables to be set in subdirectories as needed).

- `lex/Makefile`:
  - l. 28: "INCLUDES"
- `runtime/Makefile`
  - l. 82: "SUPPORTS_BYTECODE_SHARED_LIBRARIES"
  - l. 86: "SUPPORTS_NATIVE_SHARED_LIBRARIES"
  - l. 111: "NATIVE_C_SOURCESS"

**Comment by shindere** [comment]
Hello, sorry for the late response.

Just opened #8573 to fix this.

**Comment by shindere** [comment]
Fixed by #8573, closing.
